src/dotnet/commands/dotnet-msbuild/MSBuildForwardingApp.cs

```diff
@@ -36,7 +46,16 @@ public MSBuildForwardingApp(IEnumerable<string> argsToForward)
 
         public int Execute()
         {
-            return _forwardingApp.Execute();
+            try
+            {
+                Environment.SetEnvironmentVariable(TelemetrySessionIdEnvironmentVariableName, Telemetry.CurrentSessionId);
```

Comment:
Yup, we've had some bad experiences with shared state in MSBuild's unit tests, because a couple of core msbuild types get configured via env variables or read the current working directory :)